### PR TITLE
Add awareness of more override flags in bash-completion

### PR DESCRIPTION
**Pull request by sttts**
Add awareness during completion for
- `--kubeconfig`
- `--cluster`
- `--user`
- `--context`
- `--server`

TODO:
- [ ] ~~add completion for the upper flags themselves~~ (out of scope)

---

This change is [Reviewable](http://reviewable.k8s.io/reviews/kubernetes/kubernetes/23830)

### CODE REVIEW

File: pkg/kubectl/cmd/cmd.go [hunk 1]
```diff
@@ -30,30 +30,35 @@ import (
 
 const (
 	bash_completion_func = `# call kubectl get $1,
-__kubectl_namespace_flag()
+__kubectl_override_flag_list=(kubeconfig cluster user context namespace server)
+__kubectl_override_flags()
```
Comment (eparis) [inline pkg/kubectl/cmd/cmd.go]: feels like something that could be done in cobra...

Comment (eparis) [inline pkg/kubectl/cmd/cmd.go]: maybe not...    I wish it could, somehow....

Comment (sttts) [inline pkg/kubectl/cmd/cmd.go]: is that really something generic that you set the context (in the non-k8s sense) of a command? I don't really see that in cobra either.

**Comment by sttts** [comment]
@eparis ptal

**Comment by sttts** [comment]
@k8s-bot test this issue: #21990

**Comment by sttts** [comment]
@k8s-bot test this issue: #27296

**Comment by sttts** [comment]
@k8s-bot unit test this issue: #28076

**Comment by sttts** [comment]
Rebased

**Pull request merged**